app/views/simulations/new.html.erb

```diff
@@ -16,57 +16,63 @@
-              { prompt: 'Role'  },
-              class: 'c-select',
-              required: true
-            %>
-          </td>
-          <td>
-            <%= ff.select 'search_strategy', options_for_select(Simulation::SEARCH_STRATEGIES),
-              { prompt: 'Search strategy'  },
-              class: 'c-select',
-              required: true
-            %>
-          </td>
-          <td>
-            <%= ff.select 'social_strategy', options_for_select(Simulation::SOCIAL_STRATEGIES),
-              { prompt: 'Social strategy'  },
-              class: 'c-select',
-              required: true
-            %>
-          </td>
-          <td>
-            <%= number_field_tag 'simulation[users_attributes][0][quantity]', '',
-              placeholder: 1,
-              min: 1,
-              class: 'form-control simulation-quantity',
-              required: true
-            %>
-          </td>
-          <td>
-            <button type="button" class="btn btn-danger simulation-delete-row" onclick="deleteRow(event)">
-              Delete
-            </button>
-          </td>
-        </tr>
-      </tbody>
-    </table>
-    <div class="simulation-next-row">
-      <a class="btn btn-secondary simulation-next-row-button">+</a>
+    <div data-define="{table: new SimulationTable(this)}" data-context="table">
+      <table class="table">
+        <thead class="thead-inverse">
+          <tr>
+            <th>Role</th>
+            <th>Search Strategy</th>
+            <th>Social Strategy</th>
+            <th>Quantity</th>
+            <th></th>
+          </tr>
+        </thead>
+        <tbody id="simulation-table">
+          <tr>
+            <td>
+              <%= ff.select 'role', options_for_select(%w(Producer Consumer)),
+                { prompt: 'Role'  },
+                class: 'c-select',
+                required: true
+              %>
+            </td>
+            <td>
+              <%= ff.select 'search_strategy', options_for_select(Simulation::SEARCH_STRATEGIES),
+                { prompt: 'Search strategy'  },
+                class: 'c-select',
+                required: true
+              %>
+            </td>
+            <td>
+              <%= ff.select 'social_strategy', options_for_select(Simulation::SOCIAL_STRATEGIES),
+                { prompt: 'Social strategy'  },
+                class: 'c-select',
+                required: true
+              %>
+            </td>
+            <td>
+              <%= number_field_tag 'simulation[users_attributes][0][quantity]', '',
+                placeholder: 1,
+                min: 1,
+                class: 'form-control simulation-quantity',
+                required: true
+              %>
+            </td>
+            <td>
+              <%= button_tag 'Delete',
+                type: 'button',
+                class: 'btn btn-danger simulation-delete-row',
+                data: { bind_event_click: 'deleteTableRow(this)' }
```

Comment:
Here we tell twine to bind click events to the `deleteTableRow` function that's part of our sub-context (`table`).